src/resolvers/exotics/file-resolver.js

```diff
@@ -26,14 +26,23 @@ export default class FileResolver extends ExoticResolver {
 
   async resolve(): Promise<Manifest> {
     let loc = this.loc;
+    const optional = this.request.optional;
     if (!path.isAbsolute(loc)) {
       loc = path.join(this.config.cwd, loc);
     }
-    if (!(await fs.exists(loc))) {
+    if (!(await fs.exists(loc)) && !optional) {
       throw new MessageError(this.reporter.lang('doesntExist', loc));
     }
 
-    const manifest = await this.config.readManifest(loc, this.registry);
+    let manifest = await this.config.tryManifest(loc, this.registry);
+    
+    if (!manifest && optional) {
+      manifest = {
+        _registry: this.registry,
+        _loc: loc,
+        version: 'undefined',
```

Comment:
I get that, I'm just wondering: is there is a way to ignore a dependency from a resolver?